code/game/machinery/turret_control.dm

```diff
@@ -180,9 +180,32 @@
 	TC.ailock = ailock
 
 	if(istype(control_area))
+		var/lethal_only = 0 // To check if there is a lethal turret in the area.
+		var/both_mode = 0
 		for (var/obj/machinery/porta_turret/aTurret in control_area)
+			// If turret only has lethal mode - lock switching modes
+			if(aTurret.egun == 0)
+				lethal_only = 1
+				egun = 0
+			else
+				both_mode = 1
 			aTurret.setState(TC)
-
+		// If there is a turret with lethal mode only, and turret with both modes. Scan for lethal turrets, and disable them.
+		if(both_mode && lethal_only)
+			if(lethal)
+				for (var/obj/machinery/porta_turret/bTurret in control_area)
+					bTurret.setState(TC)
+				egun = 1
+			else
+				for (var/obj/machinery/porta_turret/bTurret in control_area)
+					if(bTurret.egun == 0)
+						TC.enabled = 0
```

Comment:
Race condition again. This effects all subsequent iterations. You need two instances of the `turret_checks` object to do this.